### DISCUSSION TITLE
Exposing `variables` through the `useStyles` hook

### DISCUSSION THREAD

**Discussion by marwan38** [microsoft/fluentui#23851] [Q&A]
This might be an issue on our side and misunderstanding of fluentui itself.

We've had various situations where we needed to access the component variables during the render cycle.

```ts
const {theme} = useFluentContext();
const variables = componentVariables(theme.siteVariables);

return <div>
<SomeOtherComponent variables={{color: variables.someOtherComponentInternalColorValue}} />
</div>
```

I can see that within the implementation of `useStyles` is `getStyles` that actually resolved the component variables, meaning it can be exposed through `useStyles`. My question, before thinking about patching, or putting a PR is:

1. Are we misunderstanding fluentui?
2. Has variables purposely been hidden away from `useStyles`?

Thanks

**Comment by spmonahan** [comment]
@marwan38 Can you confirm which version of Fluent you're using?

**Reply by marwan38** [reply]
Latest version: 0.63.1

**Reply by spmonahan** [reply]
@jurokapsiar This is a question for Northstar. Can you or someone from your team help out here?
